### PR TITLE
Chore: Merge main to next-release/main

### PR DESCRIPTION
**Pull request by jacoblogan**
#### Description of changes:
Bi-weekly merge of main into next-release/main

includes:
- [[Gen 2][Bugfix] Update the flutter mobile support doc.](https://github.com/aws-amplify/docs/commit/ae6c97229db06d620e391939b14361d4d48ade35)

- [chore(main): replace Twitter logo with X logo](https://github.com/aws-amplify/docs/commit/cff8db3e4cd72a5df63d503a075e4a5ce415420a)

- [Fix gen-1 storage code](https://github.com/aws-amplify/docs/commit/d84190e6101d7600b46a8cc5cdf4cef82e103188)

- [fix(main): allow code elements to work in h2/h3 headings with anchor links](https://github.com/aws-amplify/docs/commit/5d8045f9329f0048bd91caa23ac525528eabda63)

- [fix(main): allow tablist in Blockswitcher to be scrollable](https://github.com/aws-amplify/docs/commit/6ef65db960c1f117c7efad120ab69b0f03b78186)

- [chore(main): update X icon name](https://github.com/aws-amplify/docs/commit/941f6a2edebbab02b6257d94debd3689def4d7e4)

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/gen2/build-a-backend/storage/index.mdx
Comment (timngyn) [inline src/pages/gen2/build-a-backend/storage/index.mdx:1]: This shouldn't be a gen2 page right? We would need to add this to `src/pages/build-a-backend/storage` or check to see if it already exists and then make sure it has the latest updates?

Comment (jacoblogan) [reply src/pages/gen2/build-a-backend/storage/index.mdx:1]: Thanks, I've removed the gen2 page completely.  It looks like in the next-release/main branch that single page has been broken up into multiple sub pages.

Comment (timngyn) [reply src/pages/gen2/build-a-backend/storage/index.mdx:1]: I think the file got moved to here: https://github.com/aws-amplify/docs/blob/b9a31a72b7f6e17ab7f80c33490f2bd88c16a024/src/pages/%5Bplatform%5D/build-a-backend/storage/authorization/index.mdx?plain=1#L255

Comment (jacoblogan) [reply src/pages/gen2/build-a-backend/storage/index.mdx:1]: good catch, updated

**Review by timngyn — APPROVED** [review]

**Review by katiegoines — APPROVED** [review]

**Pull request merged**